### ISSUE TITLE
Alert when I'm being pinged?

### ISSUE THREAD

**Issue by kuba-orlik** [itds-consulting/android-silent-ping-sms#16]
Thank you for creating this app! I've been using it for demonstrative purposes and it blew some people's minds :D

I was wondering: if we can send silent text messages, is there a way to check if we're being pinged? Like, a notification is displayed when someone sends us a silent SMS. Or an option to block such messages. 

Is that possible?

**Comment by smarek** [comment]
Hi, nope, in ideal case this sms is consumed by baseband and the data / info about received data message is not passed through RIL to UE. 

What are the options:

  - Scanning logcat for appearances of specific messages that indicate Type 0 SMS or non-consumed data messages (drastically depends on phone type and software version) - something like this is done (poorly) in [AIMSICD](https://github.com/CellularPrivacy/Android-IMSI-Catcher-Detector)
  - Create SMS handling application (set as default SMS application for system) , but Type 0 and data-messages are usually not passed to these applications

It's not much, because the behavior is exploited in lower software/firmware levels, and mitigation would require you to have open-source-baseband or some type of low level control over closed-source-baseband (ie. disable radio or explicitly reject the silent sms delivery, so the network won't send ACK back to sender)

**Comment by kuba-orlik** [comment]
I see. One more reason to await Purism Phone ;)

**Comment by smarek** [comment]
I'm sorry, but I don't think Purism 5 Phone (as from what I've already seen) will solve this issue. Even when the phone has separated baseband from the ACPU (application / user CPU), this kind of messages (such as Type 0 or our Silent Ping SMS) will still work. Separation will only protect the user-data from being accessed through BBP (baseband) bugs (ie. stack/buffer-overflow in Qualcomm REX that will enable R/W access to phone memory), but not from direct BBP access, that result in privacy breach.

**Comment by kuba-orlik** [comment]
Yeah, but the baseband will be open source, so it can be modified to block such messages, right?

**Comment by smarek** [comment]
@kuba-orlik i think you're mistaken, Librem 5 baseband software (firmware) will not be opensource, at least I only seen the info about baseband being as much separated from main CPU as possible, and kill-switches being available. But the baseband still will be closed-source, ROM loaded (ie. the firmware will be loaded on ROM boot-up, and thus will be vulnarable.

If you have any info about Librem 5 baseband being open-sourced, please let me know, that would be exciting!

**Comment by kuba-orlik** [comment]
Oh, you're right - I somehow got the impression they're going full 110% open-source, given their history with reverse-engineering of Intel firmware and what not. It seems the baseband firmware will be ROM-loaded, but separate from the CPU. 

Still more than what's done privacy-wise on regular phones....

**Comment by smarek** [comment]
Actually, I have just written detector, stay tuned :)

**Comment by smarek** [comment]
Ok, so some development has been made, if you can build+run from GIT please do so with latest master (or tag 4-alpha-1)
This should provide you with ping detection feature!

**Comment by kuba-orlik** [comment]
Oh, that's so cool! Did you discover something new?

I only have one phone in my hand, so cannot test it atm...

**Comment by smarek** [comment]
You can actually ping yourself, ie. ping the same phone / number
